app/assets/javascripts/discourse/tests/acceptance/create-account-external-test.js

```diff
@@ -43,3 +55,24 @@ acceptance("Create Account - external auth", function (needs) {
     assert.not(exists("#new-account-username"), "it does not show the fields");
   });
 });
+
+acceptance("Create account - with associate link", function (needs) {
+  needs.hooks.beforeEach(() => {
+    setupAuthData({ associate_url: "/associate/abcde" });
+  });
+  needs.hooks.afterEach(() => {
+    document
+      .querySelector("head")
+      .removeChild(document.getElementById("data-authentication"));
+  });
+
+  test("displays associate link when allowed", async function (assert) {
+    await visit("/");
+
+    assert.ok(
+      exists("#discourse-modal div.create-account-body"),
+      "it shows the registration modal"
+    );
+    assert.ok(exists("#new-account-username"), "it shows the fields");
```

Comment:
Sorry one more comment here. It doesn't seem like we're actually testing for the associate link in this test. I see in another we're testing for the `.create-account-associate-link` class but we don't test the existence of it in any of the other tests.

Reply:
🤦 you're right, I accidentally deleted the check when refactoring. Will fix